lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -152,102 +254,55 @@ def build_pool(options)
-      pool_options[:scheme] = ssl_options && ssl_options[:enabled] ? 'https' : 'http'
-
-      if options[:user]
-        pool_options[:auth] = {:user => options[:user], :password => options[:password]}
+      if self.user
+        pool_options[:auth] = {:user => self.user, :password => self.password}
       end
 
       pool_class = ::LogStash::Outputs::ElasticSearch::HttpClient::Pool
-      pool_class.new(@logger, adapter, urls, pool_options)
+      full_urls = @options[:hosts].map {|h| host_to_url(h, client_settings[:parameters] || {}) }
+      pool_class.new(@logger, adapter, full_urls, pool_options)
     end
 
-    HOSTNAME_PORT_REGEX=/\A(?<hostname>([A-Za-z0-9\.\-]+)|\[[0-9A-Fa-f\:]+\])(:(?<port>\d+))?\Z/
-    URL_REGEX=/\A#{URI::regexp(['http', 'https'])}\z/
-    # Parse a configuration host to a normalized URL
-    def host_to_url(host, ssl=nil, path=nil, parameters=nil)
-      explicit_scheme = case ssl
-                        when true
-                          "https"
-                        when false
-                          "http"
-                        when nil
-                          nil
-                        else
-                          raise ArgumentError, "Unexpected SSL value!"
-                        end
-
-      # Ensure path starts with a /
-      if path && path[0] != '/'
-        path = "/#{path}"
-      end
-
-      url = nil
-      if host =~ URL_REGEX
-        url = URI.parse(host)
-
-        # Please note that the ssl == nil case is different! If you didn't make an explicit
-        # choice we don't complain!
-        if url.scheme == "http" && ssl == true
-          raise LogStash::ConfigurationError, "You specified a plain 'http' URL '#{host}' but set 'ssl' to true! Aborting!"
-        elsif url.scheme == "https" && ssl == false
-          raise LogStash::ConfigurationError, "You have explicitly disabled SSL but passed in an https URL '#{host}'! Aborting!"
-        end
-
-        url.scheme = explicit_scheme if explicit_scheme
-      elsif (match_results = HOSTNAME_PORT_REGEX.match(host))
-        hostname = match_results["hostname"]
-        port = match_results["port"] || 9200
-        url = URI.parse("#{explicit_scheme || 'http'}://#{hostname}:#{port}")
-      else
-        raise LogStash::ConfigurationError, "Host '#{host}' was specified, but is not valid! Use either a full URL or a hostname:port string!"
-      end
-
-      if path && url.path && url.path != "/" && url.path != ''
-        safe_url = ::LogStash::Outputs::ElasticSearch::SafeURL.without_credentials(url)
-        raise LogStash::ConfigurationError, "A path '#{url.path}' has been explicitly specified in the url '#{safe_url}', but you also specified a path of '#{path}'. This is probably a mistake, please remove one setting."
-      end
-
-      if path
-        url.path = path  # The URI library cannot stringify if it holds a nil
-      end
-
-      if parameters
+    def host_to_url(h, parameters={})
```

Comment:
That's a good idea. I think I actually just found a bug in your parameterized URLs patch where a sniffed host would *not* get the parameters appended. Combining these methods would remove that class of bug.